### PR TITLE
Added user-friendly console logging for eth1 service success + failure.

### LINKED ISSUE #1796: [Feature] Log acknowledgement of eth1 rpc endpoint success/failure
### Description
As a validator, I want to know if Teku has established a connection to the eth1 rpc endpoint I provided via the initialization log.

### Acceptance Criteria
* Log success or failure of eth1 rpc endpoint connection.

**When to occur:** 
If `--eth1-enabled=true`

### PR DESCRIPTION
**Pull request by mark-terry**
## PR Description
Added console logging to indicate when the eth1 service has caught up to chain head + present a single console message when the eth1 service is failing.

## Fixed Issue(s)

Fixes #1796 

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: pow/src/main/java/tech/pegasys/teku/pow/Eth1StatusLogger.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.TimerTask;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class Eth1StatusLogger {
+  private final AtomicBoolean timerActive = new AtomicBoolean();
+  private final AtomicInteger failCount = new AtomicInteger(0);
+  private final long interval;
+  private final TimerTask timerTask;
+  private Timer timer;
```
Comment (ajsutton) [inline pow/src/main/java/tech/pegasys/teku/pow/Eth1StatusLogger.java]: The modern version of `Timer` is `ScheduledThreadPoolExecutor` which is probably better to use.

File: pow/src/main/java/tech/pegasys/teku/pow/Eth1StatusLogger.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  public Eth1StatusLogger(final long interval) {
+    this.interval = interval;
+    this.timerTask =
+        new TimerTask() {
+          @Override
+          public void run() {
+            if (failCount.get() > 1) {
+              STATUS_LOG.eth1ServiceDown(Eth1StatusLogger.this.interval);
```
Comment (ajsutton) [inline pow/src/main/java/tech/pegasys/teku/pow/Eth1StatusLogger.java]: If we have a single request fail and everything else work, this will wind up reporting that the ETH1 service has been down for the entire timer interval which is misleading.  We really want to ignore individual errors but report when there is problem for a sustained period.  If the errors persist for more than one interval we'd also want the time reported in the logs to increase so each log reports the total outage period, not just repeatedly report the same period.

So I'd say we nee to report when requests start working to this class.  Then it can track how long requests have been failing for and alert as needed.  Having a `ErrorTrackingEth1Provider` which implements `Eth1Provider` and delegates much like `ThrottlingEth1Provider` does is probably the neatest way to do this - then it automatically tracks all the ETH1 requests we make and all the error tracking is kept in one place.

File: pow/src/main/java/tech/pegasys/teku/pow/Eth1HeadTracker.java [hunk 3]
```diff
@@ -82,6 +90,10 @@ private void onLatestBlockHead(final Block headBlock) {
     if (headAtFollowDistance
         .map(current -> current.compareTo(newHeadAtFollowDistance) < 0)
         .orElse(true)) {
+      if (!reachedHead) {
```
Comment (ajsutton) [inline pow/src/main/java/tech/pegasys/teku/pow/Eth1HeadTracker.java]: `reachedHead` will need to be an `AtomicBoolean` as multiple threads could be involved.

File: pow/src/main/java/tech/pegasys/teku/pow/DepositFetcher.java [hunk 4]
```diff
@@ -94,11 +97,12 @@ public DepositFetcher(
         .depositEventInRange(fromBlock, toBlock)
         .exceptionallyCompose(
             (err) -> {
-              LOG.warn(
+              LOG.debug(
                   "Failed to request deposit events for block numbers in the range ({}, {}). Retrying.",
                   fromBlockNumber,
                   toBlockNumber,
                   err);
+              eth1StatusLogger.fail();
```
Comment (ajsutton) [inline pow/src/main/java/tech/pegasys/teku/pow/DepositFetcher.java]: Couldn't this also be handled by `ErrorTrackignEth1Provider`? It can just use the `catchAndRethrow` method of `SafeFuture`.

File: pow/src/main/java/tech/pegasys/teku/pow/Eth1HeadTracker.java [hunk 5]
```diff
@@ -82,6 +90,10 @@ private void onLatestBlockHead(final Block headBlock) {
     if (headAtFollowDistance
         .map(current -> current.compareTo(newHeadAtFollowDistance) < 0)
         .orElse(true)) {
+      if (!reachedHead.get()) {
```
Comment (ajsutton) [inline pow/src/main/java/tech/pegasys/teku/pow/Eth1HeadTracker.java]:
```suggestion
      if (reachedHead.compareAndSet(false, true)) {
```

Then drop the `reachedHead.set(true)`.  That way you're using a single atomic operation to check and update so it's thread safe.

File: services/powchain/src/main/java/tech/pegasys/teku/services/powchain/PowchainService.java [hunk 6]
```diff
@@ -41,6 +43,7 @@
 
   private final Eth1DepositManager eth1DepositManager;
   private final Eth1HeadTracker headTracker;
+  private final Eth1StatusLogger eth1StatusLogger = new Eth1StatusLogger();
```
Comment (ajsutton) [inline services/powchain/src/main/java/tech/pegasys/teku/services/powchain/PowchainService.java]: Probably should make this just a local variable.  Ultimately we should only need to provide it to `ErrorTrackingEth1Provider`.

File: pow/src/main/java/tech/pegasys/teku/pow/Eth1StatusLogger.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    this.executor = Executors.newScheduledThreadPool(1);
+    scheduledFuture =
+        executor.scheduleAtFixedRate(
+            () ->
+                STATUS_LOG.eth1ServiceDown(
+                    Duration.between(startInstant, Instant.now()).getSeconds()),
+            LOG_INTERVAL,
+            LOG_INTERVAL,
+            TimeUnit.MILLISECONDS);
+  }
```
Comment (ajsutton) [inline pow/src/main/java/tech/pegasys/teku/pow/Eth1StatusLogger.java]: Sorry, I led you astray here - don't know why I didn't think of this when I mentioned `ScheduledExecutorService`.  We have an abstraction (`AsyncRunner`) that makes running things after some period a lot more testable and generally makes managing life cycle easier.  And we also have one for accessing time - `TimeProvider` (which you can get from the `ServiceConfig` passed into `PowchainService` constructor).

So using those I think we wind up with:
```
public class Eth1StatusLogger {
  private static final Logger LOG = LogManager.getLogger();
  private static final int LOG_INTERVAL = 30000;

  private final AsyncRunner asyncRunner;
  private final TimeProvider timeProvider;

  private Optional<Cancellable> activeReporter = Optional.empty();

  public Eth1StatusLogger(final AsyncRunner asyncRunner, final TimeProvider timeProvider) {
    this.asyncRunner = asyncRunner;
    this.timeProvider = timeProvider;
  }

  synchronized void success() {
    activeReporter.ifPresent(
        reporter -> {
          reporter.cancel();
          activeReporter = Optional.empty();
        });
  }

  synchronized void fail() {
    if (activeReporter.isEmpty()) {
      final UnsignedLong outageStartInSeconds = timeProvider.getTimeInSeconds();
      final Cancellable reporter =
          asyncRunner.runWithFixedDelay(
              () -> reportOutage(outageStartInSeconds),
              LOG_INTERVAL,
              TimeUnit.MILLISECONDS,
              error -> LOG.error("Failed to check ETH1 status", error));
      activeReporter = Optional.of(reporter);
    }
  }

  private void reportOutage(final UnsignedLong outageStartInSeconds) {
    STATUS_LOG.eth1ServiceDown(timeProvider.getTimeInSeconds().minus(outageStartInSeconds));
  }
}
```

So basically when a failure comes in, start a periodic report of how long the failure has been happening and when success happens cancel that reporting.

File: pow/src/main/java/tech/pegasys/teku/pow/Eth1StatusLogger.java [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+  synchronized void fail() {
+    if (activeReporter.isEmpty()) {
+      final UnsignedLong outageStartInSeconds = timeProvider.getTimeInSeconds();
+      final Cancellable reporter =
+          asyncRunner.runWithFixedDelay(
+              () -> reportOutage(outageStartInSeconds),
+              LOG_INTERVAL,
+              TimeUnit.MILLISECONDS,
+              error -> LOG.error("Failed to check ETH1 status", error));
```
Comment (macfarla) [inline pow/src/main/java/tech/pegasys/teku/pow/Eth1StatusLogger.java]: nit: Eth1 is used elsewhere rather than ETH1

File: pow/src/main/java/tech/pegasys/teku/pow/Eth1HeadTracker.java [hunk 9]
```diff
@@ -56,7 +58,7 @@ private void pollLatestHead() {
         .thenAccept(this::onLatestBlockHead)
         .exceptionally(
             error -> {
-              LOG.warn("Failed to get latest ETH1 chain head. Will retry.", error);
+              LOG.debug("Failed to get latest ETH1 chain head. Will retry.", error);
```
Comment (macfarla) [inline pow/src/main/java/tech/pegasys/teku/pow/Eth1HeadTracker.java]: Nit: Eth1 used elsewhere rather than ETH1

File: pow/src/main/java/tech/pegasys/teku/pow/Eth1HeadTracker.java [hunk 10]
```diff
@@ -82,6 +84,10 @@ private void onLatestBlockHead(final Block headBlock) {
     if (headAtFollowDistance
         .map(current -> current.compareTo(newHeadAtFollowDistance) < 0)
         .orElse(true)) {
+      if (reachedHead.compareAndSet(false, true)) {
+        STATUS_LOG.eth1AtHead();
+        reachedHead.set(true);
+      }
       headAtFollowDistance = Optional.of(newHeadAtFollowDistance);
       LOG.debug("ETH1 block at follow distance updated to {}", newHeadAtFollowDistance);
```
Comment (macfarla) [inline pow/src/main/java/tech/pegasys/teku/pow/Eth1HeadTracker.java:92]: Nit: Eth1 used elsewhere rather than ETH1

**Review by macfarla — APPROVED** [review]
LGTM - would be nice to use "Eth1" consistently in log messages if that's the preferred version (rather than "ETH1")

**Review by ajsutton — APPROVED** [review]
LGTM.
